File: app/src/main/java/cc/metapro/openct/grades/cet/CETService.java
package cc.metapro.openct.grades.cet;

/*
 *  Copyright 2016 - 2017 OpenCT open source class table
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Query;

public interface CETService {

    @GET("http://www.chsi.com.cn/cet/query")
    Call<String> queryCET(
            @Header("Referer") String referer,
            @Query("zkzh") String num,
            @Query("xm") String name,
            @Query("_t") String t
    );
}

File: app/src/main/java/cc/metapro/openct/data/service/CETService.kt
/*
 *  Copyright 2016 - 2017 OpenCT open source class table
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package cc.metapro.openct.data.service

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Query


interface CETService {

    @GET("http://www.chsi.com.cn/cet/query")
    fun queryCET(@Header("Referer") refer: String,
                 @Query("zkzh") num: String,
                 @Query("xm") name: String,
                 @Query("_t") t: String
    ): Call<String>
}